src/Tasks/Touch.cs

```diff
@@ -185,9 +221,16 @@ public override bool Execute()
                     return false;
                 }
             }
+
+            // Ignore touching the disk when FailIfNotIncremental.
+            if (FailIfNotIncremental)
```

Comment:
I'm wondering if we should take into account some of the failure cases below before erroring here.

Reply:
Question mode would stop before any write action.  Those failure modes are afterwards that makes it awkward to support.  
I am fine with Question mode only checking would write than can write.